addons/pos_restaurant/__init__.py

```diff
@@ -2,3 +2,7 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from . import models
+
+def _pos_restaurant_post_init(env):
+    if env['ir.module.module'].search_count([('state', '=', 'installed'), ('name', '=', 'pos_preparation_display')], limit=1):
+        env['pos_preparation_display.display']._setup_default_data_for_main_pos_restaurant()
```

Comment:
I'm not really aware of how exactly odoo modules are loaded, but pos_preparation_display and pos_restaurant both depend on point_of_sale, therefore I assume there is no reason for one to be loaded before another (maybe there are sequences that can be defined, but personnaly I'm not fan of using that). And the method _setup_default_data_for_main_pos_restaurant must be called when both modules are loaded (and particularly their data), so I put that post_init for both. I also put it for point_of_sale just in case but it is probably useless.

Reply:
From how I understand it, `_setup_default_data_for_main_pos_restaurant` depends on two records, 1 from `pos_restaurant` and the second one from `pos_preparation_display`. Since the `pos_restaurant_preparation_display` addon depends on both `pos_restaurant` and `pos_preparation_display`, you have the assurance that `_setup_default_data_for_main_pos_restaurant` can be called in that addon.

Reply:
Ok thank you